### PR TITLE
Ca2250

### PR DESCRIPTION
**Pull request by NewellClark**
## Summary

Add docs for [CA2250](https://github.com/dotnet/roslyn-analyzers/pull/4864/)

[Tracking issue](https://github.com/dotnet/docs/issues/23900)

### CODE REVIEW

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+dev_langs:
+- CSharp
+- VB
+---
+# CA2250: Use `ThrowIfCancellationRequested`
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2250|
+| **Category** |[Microsoft.Usage](usage-warnings.md)|
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:22]: Why the `Microsoft.` prefix? isn't it just "Usage"?

```suggestion
| **Category** |[Microsoft.Usage](usage-warnings.md)|
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+To fix violations, replace the conditional statement with a call to <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested()>.
+
+```csharp
+using System;
+using System.Threading;
+
+public void MySlowMethod(CancellationToken token)
+{
+    //  Violation
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:43]:
```suggestion
    // Violation
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    if (token.IsCancellationRequested)
+        throw new OperationCanceledException();
+    
+    //  Fix
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:47]:
```suggestion
    // Fix
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    token.ThrowIfCancellationRequested();
+    
+    //  Violation
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:50]:
```suggestion
    // Violation
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    if (token.IsCancellationRequested)
+        throw new OperationCanceledException();
+    else
+        DoSomethingElse();
+    
+    //  Fix
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:56]:
```suggestion
    // Fix
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+}
+```
+
+```vb
+Imports System
+Imports System.Threading
+
+Public Sub MySlowMethod(token As CancellationToken)
+
+    'Violation
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:68]:
```suggestion

    ' Violation
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    If token.IsCancellationRequested Then
+        Throw New OperationCanceledException()
+    End If
+    
+    'Fix
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:73]:
```suggestion
    ' Fix
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    token.ThrowIfCancellationRequested()
+    
+    'Violation
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:76]:
```suggestion
    ' Violation
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    If token.IsCancellationRequested Then
+        Throw New OperationCanceledException()
+    Else
+        DoSomethingElse()
+    End If
+    
+    'Fix
```
Comment (Youssef1313) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:83]:
```suggestion
    ' Fix
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+| **Category** |[Usage](usage-warnings.md)|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+This rule flags conditional statements that check <xref:System.Threading.CancellationToken.IsCancellationRequested> before throwing <xref:System.OperationCanceledException>.
+
+## Rule description
+
+You can accomplish the same thing by calling <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested()>.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:31]:
```suggestion
You can accomplish the same thing by calling <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested?displayProperty=nameWithType>.
```

File: docs/fundamentals/code-analysis/quality-rules/ca2250.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+## How to fix violations
+
+To fix violations, replace the conditional statement with a call to <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested()>.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/quality-rules/ca2250.md:35]:
```suggestion
To fix violations, replace the conditional statement with a call to <xref:System.Threading.CancellationToken.ThrowIfCancellationRequested>.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**